src/sentry/static/sentry/app/stores/groupingStore.jsx

```diff
@@ -0,0 +1,368 @@
+        });
+
+        this.loading = false;
+        this.error = false;
+        this.triggerFetchState();
+      },
+      () => {
+        this.loading = false;
+        this.error = true;
+        this.triggerFetchState();
+      }
+    );
+  },
+
+  // Toggle merge checkbox
+  onToggleMerge(id) {
+    let checked;
+
+    // Don't do anything if item is busy
+    let state = this.mergeState.has(id) && this.mergeState.get(id);
+    if (!state || state.busy !== true) {
+      if (this.mergeList.has(id)) {
+        this.mergeList.delete(id);
+        checked = false;
+      } else {
+        this.mergeList.add(id);
+        checked = true;
+      }
+
+      this.setStateForId(this.mergeState, id, {
+        checked
+      });
+
+      this.triggerMergeState();
+    }
+  },
+
+  // Toggle unmerge check box
+  onToggleUnmerge(id) {
+    let checked;
+
+    // Uncheck an item to unmerg
+    let state = this.unmergeState.has(id) && this.unmergeState.get(id);
+    if (!state || state.busy !== true) {
+      if (this.unmergeList.has(id)) {
+        this.unmergeList.delete(id);
+        checked = false;
+
+        // If there was a single unchecked item before, make sure we reset its disabled state
+        if (this.remainingItem) {
+          this.setStateForId(this.unmergeState, this.remainingItem.id, {
+            disabled: false
+          });
+          this.remainingItem = null;
+        }
+      } else {
+        // at least 1 item must be unchecked for unmerge
+        // make sure that not all events have been selected
+
+        // Account for items in unmerge queue, or "locked" items
+        let lockedItems = Array.from(this.unmergeState.values()).filter(
+          ({busy}) => busy
+        ) || [];
+
+        if (this.unmergeList.size + 1 < this.mergedItems.length - lockedItems.length) {
+          this.unmergeList.add(id);
+          checked = true;
+
+          // Check if there's only one remaining item, and make sure to disable it from being
+          // selected to unmerge
+          if (
+            this.unmergeList.size + 1 ===
+            this.mergedItems.length - lockedItems.length
+          ) {
+            let remainingItem = this.mergedItems.find(item => {
+              let notSelected = !this.unmergeList.has(item.id);
+              let itemState =
+                this.unmergeState.has(item.id) && this.unmergeState.get(item.id);
+              return notSelected && (!itemState || !itemState.busy);
+            });
+            if (remainingItem) {
+              this.remainingItem = remainingItem;
+              this.setStateForId(this.unmergeState, remainingItem.id, {
+                disabled: true
+              });
+            }
+          }
+        }
+      }
+
+      // Update "checked" state for row
+      this.setStateForId(this.unmergeState, id, {
+        checked
+      });
+
+      this.triggerUnmergeState();
+    }
+  },
+
+  onUnmerge({groupId, loadingMessage, successMessage, errorMessage}) {
```

Comment:
Aside: one of the things I really dislike about Reflux is that you can't "search" for usage of `onUnmerge` – since this is a generated name derived from the `merge` action. At first I was really confused about where these methods were called until I remembered how Reflux works.